tools/stonith_admin.c

```diff
@@ -382,44 +364,30 @@ handle_history(stonith_t *st, const char *target, int timeout, int quiet,
             continue;
         }
 
-        if (hp->state == st_failed) {
-            action_s = fence_action_str(hp->action);
-            printf("%s failed to %s node %s on behalf of %s from %s at %s\n",
-                   hp->delegate ? hp->delegate : "We", action_s, hp->target,
-                   hp->client, hp->origin, ctime(&complete));
-
-        } else if (hp->state == st_done) {
-            print_fence_event(latest);
-
-        } else {
-            /* ocf:pacemaker:controld depends on "wishes to" being
-             * in this output, when used with older versions of DLM
-             * that don't report stateful_merge_wait
-             */
-            action_s = fence_action_str(hp->action);
-            printf("%s at %s wishes to %s node %s - %d %lld\n",
-                   hp->client, hp->origin, action_s, hp->target, hp->state,
-                   (long long) complete);
-        }
-
-        free(action_s);
+        out->stonith_event(out, hp);
     }
 
     if (latest) {
         if (quiet) {
-            printf("%lld\n", (long long) latest->completed);
-        } else if (!verbose) { // already printed if verbose
-            print_fence_event(latest);
+            char *buf = crm_itoa(latest->completed);
+            out->item(out, "Completed", buf);
```

Comment:
If quiet is true, this is going to /dev/null :)

BTW there is one more quirk with quiet. Our general approach is that XML output ignores any quiet/verbose settings and instead always prints the same (full) level of detail. I'd expect the same will be true of any machine-readable format we add in the future. HTML is questionable, but since modern CSS can collapse info, I'd lean to full detail there too.

Quiet is mainly intended for ease of scripting. In one sense, this project obsoletes it, since the XML output will be more reliable for higher-level tools to parse. But there may still be a use for quiet, for simple shell scripting where XML parsing is overkill.

I'm thinking the quiet model can be supported by keeping the "if quiet" guards for everything, and setting quiet to false early on if the output type is anything but text. Maybe we could abstract that into a function so the caller can do something like: if !format_supports_quiet(out) quiet = false

In that case, we don't really need item(), we can just use info().